functions/Export-DbaAvailabiltyGroup.ps1

```diff
@@ -0,0 +1,173 @@
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. To use:
+
+$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter. 
+
+SQL Server does not accept Windows credentials being passed as credentials. To connect as a different Windows user, run PowerShell as that user.
+
+.NOTES 
+Author: Chrissy LeMaire (@cl), netnerds.net
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Export-DbaAvailabilityGroup
+
+.EXAMPLE
+Export-DbaAvailabilityGroup -SqlServer sql2012 -FilePath 'C:\temp\availability_group_exports'
+
+Exports all Availability Groups from SQL server "sql2012". Output scripts are witten to the C:\temp\availability_group_exports directory.
+
+.EXAMPLE
+Export-DbaAvailabilityGroup -SqlServer sql2012 -FilePath 'C:\temp\availability_group_exports' -AvailabilityGroups AG1,AG2
+
+Exports Availability Groups AG1 and AG2 from SQL server "sql2012". Output scripts are witten to the C:\temp\availability_group_exports directory.
+
+.EXAMPLE
+Export-DbaAvailabilityGroup -SqlServer sql2014 -FilePath 'C:\temp\availability_group_exports' -NoClobber
+
+Exports all Availability Groups from SQL server "sql2014". Output scripts are witten to the C:\temp\availability_group_exports directory. If the export file already exists it will not be overwritten.
+
+.NOTES 
+Author: Chris Sommer (@cjsommer), cjsommmer.com
+
+.LINK 
+https://dbatools.io/Export-DbaAvailabilityGroup
+
+#>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		    [Alias("ServerInstance", "SqlInstance")]
+		    [object[]]$SqlServer,
+
+		[Alias("OutputLocation", "Path")]
+		    [string]$FilePath,
+
+        [switch]$NoClobber ,
+
+		[object]$SqlCredential
+	)
+
+	DynamicParam { if ($sqlserver) { return Get-ParamSqlAvailabilityGroups -SqlServer $sqlserver[0] -SqlCredential $SqlCredential } }
+	
+    BEGIN
+    {
+        Write-Output "Beginning Export-DbaAvailabilityGroup"
+        $AvailabilityGroups = $PSBoundParameters.AvailabilityGroups
+    }
+
+    PROCESS
+    {
+        $SQLObj = New-Object "Microsoft.SqlServer.Management.Smo.Server" $SQLServer
+        $SQLObj.ConnectionContext.Connect()
```

Comment:
We are still working on naming conventions, but please use $server for now instead of $sqlobj. In general, mentioning the datatype in the varaiblen ame is not encouraged in PowerShell.